src/App.tsx

```diff
@@ -15,36 +23,103 @@
   'Garlic',
 ];
 
+const sortByLength = (arr: Goods, isReversed: boolean) => {
+  return arr.toSorted((a: string, b: string) => {
+    if (isReversed) {
+      return b.length - a.length;
+    }
+
+    return a.length - b.length;
+  });
+};
+
+const sortAlphabetically = (arr: Goods, isReversed: boolean) => {
+  if (isReversed) {
+    return arr.toSorted().reverse();
+  }
+
+  return arr.toSorted();
+};
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Goods>(goodsFromServer);
+  const [sorting, setSorting] = useState<SortTypes | null>(null);
+  const [isReversed, setIsReveresed] = useState(false);
+
+  const handleSortAlphabettically = () => {
+    setGoods((g: Goods) => {
+      return sortAlphabetically(g, isReversed);
+    });
+    setSorting(SortTypes.ALPHABETICAL);
+  };
+
+  const handleSortByLength = () => {
+    setGoods((g: Goods) => {
+      return sortByLength(g, isReversed);
+    });
+    setSorting(SortTypes.LENGTH);
+  };
+
+  const handleReverse = () => {
+    setGoods((g: Goods) => {
+      return [...g].reverse();
+    });
+    setIsReveresed(!isReversed);
```

Comment:
There is a typo in the state update function `setIsReveresed`. It should be `setIsReversed` to correctly update the state variable.